src/systemTest/java/com/flightstats/hub/callback/CacheObject.java

```diff
@@ -0,0 +1,33 @@
+package com.flightstats.hub.callback;
+
+import com.flightstats.hub.callback.model.RequestObject;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+public class CacheObject {
+
+    private static final Map<String, List<String>> cacheRequestObject = new ConcurrentHashMap<>();
+
+    public void put(RequestObject requestObject) {
+        String webhookName = requestObject.getName();
+
+        synchronized (CacheObject.class) {
+            if (cacheRequestObject.containsKey(webhookName)) {
```

Comment:
I think this can be replaced with computeIfAbsent() and a concurrent safe list. In which case this entire method boils down to the following. Which is thread safe because both the hashmap and lists are concurrent safe versions.
```
public put(RequestObject requestObject) {
        String webhookName = requestObject.getName();
        cacheRequestObject.computeIfAbsent(webhookName, emptyList -> new CopyOnWriteArrayList()).addAll(requestObject.getUris());
}
```

An example:
```
        ConcurrentHashMap<String,List<String>> m = new ConcurrentHashMap<>();
        m.computeIfAbsent("foo", emptyList -> new CopyOnWriteArrayList<>()).addAll(Arrays.asList("a", "b"));
        m.computeIfAbsent("foo", emptyList -> new CopyOnWriteArrayList<>()).addAll(Arrays.asList("c", "d"));
        System.out.println(m.values());
```
Output is ```[[a, b, c, d]]```

Reply:
That looks lot nicer.  In this case, there will put multiple times for the same key. Similar to Multimap.